src/components/Activity/ActivityCard.tsx

```diff
@@ -160,16 +161,22 @@ const ActivityCard = ({
               overflow="hidden"
               whiteSpace="nowrap"
               textOverflow="ellipsis"
-              w={{ base: '50%', md: '70%' }}
+              onClick={() => router.push(activityCardLinkRoute)}
```

Comment:
This breaks accessibility, either use a Link, or add accessible link attributes.
```suggestion
              onClick={() => router.push(activityCardLinkRoute)}
              tabindex="0" 
              role="link"
```